crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -320,6 +320,10 @@ where
     ) -> Result<BlockStatus, InsertBlockErrorKind> {
         debug_assert!(self.validate_block(&block).is_ok(), "Block must be validated");
 
+        if self.is_block_hash_canonical(&block.hash())? {
+            return Ok(BlockStatus::Valid(BlockAttachment::Canonical))
+        }
```

Comment:
the block has been validated at this point, unclear why this check is needed

Reply:
i was not clearing the buffered block that was used as pipeline target, removed this check